tests/i18n/tests.py

```diff
@@ -1472,12 +1472,32 @@ def test_parse_language_cookie(self):
         r.META = {'HTTP_ACCEPT_LANGUAGE': 'de'}
         self.assertEqual(g(r), 'zh-hans')
 
+    @override_settings(
+        LANGUAGES=[
+            ('en', 'English'),
+            ('de', 'German'),
+            ('de-at', 'Austrian German'),
+            ('pl', 'Polish'),
+        ]
+    )
     def test_get_language_from_path_real(self):
         g = trans_real.get_language_from_path
         self.assertEqual(g('/pl/'), 'pl')
         self.assertEqual(g('/pl'), 'pl')
         self.assertIsNone(g('/xyz/'))
 
+        self.assertEqual(g('/en/'), 'en')
+        self.assertEqual(g('/en-gb/'), 'en')
+
+        self.assertEqual(g('/de/'), 'de')
+        self.assertEqual(g('/de-at/'), 'de-at')
+        self.assertEqual(g('/de-ch/'), 'de')
+
+        # Regresion #27063: Make sure that pages starting with /de* dont change language to german
+        self.assertIsNone(g('/de-simple-page/'))
```

Comment:
What I meant about splitting the tests in a separate commit only applies to the tests that already pass before the fix. The tests that are fixed by the change should go in the second commit along with the fix.
Actually, I only see this assertion failing before the fix, so it might be that the other tests need adjusting so that they're proper regression tests: `self.assertIsNone(g('/de-simple-page/'))`